hca/util/__init__.py

```diff
@@ -283,9 +283,14 @@ def __init__(self, config=None, swagger_url=None, **session_kwargs):
         self.methods = {}
         self.commands = [self.login, self.logout, self.refresh_swagger]
         self.http_paths = collections.defaultdict(dict)
-        self.host = "{scheme}://{host}{base}".format(scheme=self.scheme,
-                                                     host=self.swagger_spec["host"],
-                                                     base=self.swagger_spec["basePath"])
+        if "openapi" in self.swagger_spec:
```

Comment:
Nit: Would it be possible to have a variable similar to `USING_PYTHON2` specifying the swagger version?

Reply:
I don't think hardcoding the version of the API spec language in each client is necessary/good, because it makes upgrading the server-side version harder. Instead the client should ideally detect the version of the server-supplied spec and adjust accordingly (which is what is done here).

I do agree that version-specific behavior could be better separated in this class. In the interest of expediency I just put a few if statements here, but ideally we would separate the code more thoroughly or even factor it out into an "OpenAPISpecParser" and "SwaggerAPISpecParser" or something.